### ISSUE TITLE
OIDC / OAuth2 - add end_session_endpoint to the discovery data

### ISSUE THREAD

**Issue by yaleman** [kanidm/kanidm#1997]
**Is your feature request related to a problem? Please describe.**

Add the `end_session_endpoint` field to the OIDC discovery endpoint.

**Describe the solution you'd like**

`end_session_endpoint` needs a URL in [OidcDiscoveryResponse](https://github.com/kanidm/kanidm/blob/f6001504a9ab911f5c5ab77cbf1ea21ed9d563e0/proto/src/oauth2.rs#L289).

The endpoint was added in #1184 and is `{base_url}/oauth2/token/revoke`

**Additional context**

* https://openid.net/specs/openid-connect-rpinitiated-1_0.html#OPMetadata

**Comment by Firstyear** [comment]
I thought that the standad oauth2 revoke was different to the oidc end session endpoint?

**Comment by yaleman** [comment]
It looks like we can just point them to the Kanidm logout URL as a bare minimum:

> 2.  [RP-Initiated Logout](https://openid.net/specs/openid-connect-rpinitiated-1_0.html#RPLogout)
> 
> An RP requests that the OP log out the End-User by redirecting the End-User's User Agent to the **OP's Logout Endpoint**. This URL is normally obtained via the `end_session_endpoint` element of the OP's Discovery response or may be learned via other mechanisms.

There's additional optional/recommended fields, but we can largely ignore them for the MVP support.

... or we could add a route `/oauth2/logout` (which triggers logout) to support the MVP and build functionality from there.

**Comment by yaleman** [comment]
Removed this from the original notes because I misread the spec:

> The endpoint was added in #1184 and is `{base_url}/oauth2/token/revoke`

**Comment by Firstyear** [comment]
Yeah, I think this needs to go to a dedicated signout page. I think the "gotcha" here is this will trigger a full sign out of the whole session, not just of the singular oidc application. At least, that's how it reads from the oidc spec anyway. So I think we need to decide if we do the full sign out, or just the single app signout.

**Comment by yaleman** [comment]
Yeah with the provision of an `id_token_hint` at the start of the session it is supposed to be able to support round-tripping the single-session cancellation, but that's a TODO in our code currently (and a whole thing in itself).
